c2corg_ui/static/js/searchfilters.js

```diff
@@ -0,0 +1,144 @@
+goog.provide('app.searchFiltersDirective');
+goog.provide('app.SearchFiltersController');
+
+goog.require('app');
+
+
+/**
+ * Provides the 'searchFiltersDirective' directive, which is used to enrich
+ * filtering components in the documents listing pages.
+ * 
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.searchFiltersDirective = function() {
+  return {
+    restrict: 'A',
+    controller: 'appSearchFiltersController',
+    bindToController: true,
+    scope: true,
+    controllerAs: 'searchFiltersCtrl',
+    link: function(scope, element, attrs, ctrl) {
+      // Init sliders
+      $('.range-between').slider({min: 0, max: 9999, value: [0, 9999]});
+      
+      var elevationFilter = $('#elevation-filter');
+      var heightFilter = $('#height-filter');
+
+      // Update elevation slider values
+      elevationFilter.on('slide', function(slideEvt) {
+        ctrl.elevation.min = slideEvt.value[0];
+        ctrl.elevation.max = slideEvt.value[1];
+        scope.$apply();
+      });
+
+      // Update height diff slider values
+      heightFilter.on('slide', function(slideEvt) {
+        ctrl.height_diff_up.min = slideEvt.value[0];
+        ctrl.height_diff_up.max = slideEvt.value[1];
+        scope.$apply();
+      });
+    }
+  }
+};
+
+
+app.module.directive('appSearchFilters', app.searchFiltersDirective);
+
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {angular.JQLite} $element Element.
+ * @param {angular.Attributes} $attrs Attributes.
+ * @constructor
+ * @ngInject
+ * @export
+ */
+app.SearchFiltersController = function($scope, $element, $attrs) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  // FIXME: not generic to all document types
+  // Init filters object
+  angular.extend(this.scope_,  {
+    activities : [], waypoint_types : [], culture : null, around : null
+  });
+
+  /**
+   * @type {Object}
```

Comment:
I don't think that is worth the time doing the typedef here because those parameters are temporary anyway and should be reworked when generalizing the directive for all doc types.